File: leetcode/3895.py
class Solution:
    def countDigitOccurrences(self, nums: list[int], digit: int) -> int:
        def count_digits(n, cnt = 0):
            if n == 0:
                return cnt
            return count_digits(n // 10, cnt + 1 if digit == n % 10 else cnt)

        def f(acc, a):
            return acc + count_digits(a)

        return reduce(f, nums, 0)

File: leetcode/3895.kt
class Solution {
    fun countDigitOccurrences(nums: IntArray, digit: Int): Int {
        val compare = digit.toString()[0]
        var cnt = 0
        for (n in nums) {
            for (c in n.toString()) {
                if (c == compare) cnt += 1
            }
        }
        return cnt
    }
}
